CLI: Force use of NPM during bootstrap of CRA repros

gaetanmaisse opened the pull request:
Issue: CRA E2E tests are crashing because CRA app isn't bootstrap properly

## What I did

As of CRA 5 `--use-npm` option has been removed.
To force usage of NPM the only hacky way I found is to override `npm_config_user_agent` env var.

See related PR: https://github.com/facebook/create-react-app/pull/11322
And changelog: https://github.com/facebook/create-react-app/releases/tag/v5.0.0

---

Fixing this highlighted a dependency issue, I will investigate on this:
![image](https://user-images.githubusercontent.com/4112568/150126934-a63a355b-9bc9-4c92-803f-d843d2966d73.png)


## How to test

- [ ] CRA e2e should be in better shape

Before:
![image](https://user-images.githubusercontent.com/4112568/150119656-745c18fb-700b-4308-8503-2012593cd6c1.png)


After:
![image](https://user-images.githubusercontent.com/4112568/150119681-7d61ddd1-ed25-4f6d-93e6-fd0a792f3b73.png)

gaetanmaisse committed:
fix: force use of NPM during bootstrap of CRA repros

As of CRA 5 `--use-npm` option has been removed.
To force usage of NPM the only hacky way I found is to override `npm_config_user_agent` env var.

See related PR: https://github.com/facebook/create-react-app/pull/11322
And changelog: https://github.com/facebook/create-react-app/releases/tag/v5.0.0

yannbf commented:
Hey @gaetanmaisse nice stuff! Unfortunately the CI is failing (for another reason):

```
ERR! Error: @storybook/core-server tried to access @storybook/manager-webpack5 (a peer dependency) but it isn't provided by its ancestors; this makes the require call ambiguous and unsound.
ERR! 
ERR! Required package: @storybook/manager-webpack5 (via "@storybook/manager-webpack5/prebuilt-manager")
ERR! Required by: @storybook/core-server@virtual:6abf09ca7260122d399a04c20ef0ce51909f1b2b123434f7b32e38baf8bcb6fbdea23748f97273cdd258e7553f037cc5070fa5a5a80ad6784bf2b5e6245d6737#npm:6.5.0-alpha.21 (via /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-a7f7549b89/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.21-6e6ca83241-8.zip/node_modules/@storybook/core-server/dist/cjs/)
ERR! 
ERR! Ancestor breaking the chain: @storybook/core@virtual:3091a69b9af0dff150eb03a2bfd1fe74e6c79caeda2b6d8df8e148ef1c852973027895b513f14285db6d7b6d51c443e213cea03cf3421913d3f390a88dc481c8#npm:6.5.0-alpha.21
ERR! 
ERR! 
ERR! Require stack:
```

gaetanmaisse commented:
hey @yannbf  👋🏻  I've noticed it too, I will try to investigate this one tonight 😃

gaetanmaisse committed:
fix: add missing peer dep in @storybook/core

As `@storybook/manager-webpack5` is listed as peerDep in `@storybook/core-server` we need to add it to `@storybook/core` also.
